include/real/boundary_helper.hpp

```diff
@@ -22,7 +23,7 @@ namespace boost {
              * @return a bool that is true if and only if, the number represented by lsh is lower than
              * the represented number by rhs.
              */
-            bool aligned_vectors_is_lower(const std::vector<int> &lhs, const std::vector<int> &rhs) {
+            static bool aligned_vectors_is_lower(const std::vector<int> &lhs, const std::vector<int> &rhs) {
```

Comment:
Previously, all of these methods were floating around in a namespace, which led to issues that were discussed previously in email, with respect to linkage. I encapsulated these methods in classes, then made the methods static because it's not attached to a particular object.